Foundation/NSPathUtilities.swift

```diff
@@ -345,7 +345,100 @@ public extension NSString {
     /// - Experiment: This is a draft API currently under consideration for official import into Foundation
     /// - Note: Since this API is under consideration it may be either removed or revised in the near future
     public func completePathIntoString(inout outputName: NSString?, caseSensitive flag: Bool, inout matchesIntoArray outputArray: [NSString], filterTypes: [String]?) -> Int {
```

Comment:
so this method should handle a few cases: what if it starts with a "~"? consider both the case of "~/" and "~"
also isn't there a fast path case? where the input is ""